### ISSUE TITLE
Zoomlevel 0, not becoming idle

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#6667]
When looking into the Android CI tests being flaky in #6546. I was able to pinpoint that a map at zoomlevel 0 keeps using CPU resources and allocating memory while it should be in an idle state (as it is with other zoomlevels).
### Analysis

App startup, not showing a map:
<img width="1440" alt="screen shot 2016-10-12 at 09 53 07" src="https://cloud.githubusercontent.com/assets/2151639/19301884/fc061c2a-9061-11e6-93f6-4bff60c54657.png">

Open screen with a map at zoom 0:
<img width="1012" alt="screen shot 2016-10-12 at 10 02 05" src="https://cloud.githubusercontent.com/assets/2151639/19302211/81c444c6-9063-11e6-8dd7-5d0236460a6f.png">

Double tap the map to go to zoom 1:
<img width="1029" alt="screen shot 2016-10-12 at 10 06 32 1" src="https://cloud.githubusercontent.com/assets/2151639/19302235/95d4b1f8-9063-11e6-9616-a65adbf59d03.png">
### Impact on CI Espresso tests

The Android espresso testing framework is smart enough to wait until the app under test becomes idle (waits for message qeue to be empty and threads to halt). When the test app is idle, it will check if our map has been correctly loaded and notifies that the tests can run.

I noticed that tests showing zoomlevel 0, never start. It seems the app never returns to an idle state. When this occurs, the application stays open for 45 seconds until it espresso times out. Strange behaviour here is if I double tap while the application is timing out the tests will run.

**Comment by tobrun** [comment]
Tried looking into what is happening that keeps the CPU active using a 10 second trace:

<img width="1100" alt="screen shot 2016-10-12 at 13 37 20" src="https://cloud.githubusercontent.com/assets/2151639/19308725/63670856-9081-11e6-9193-8355c558f1c7.png">

When you zoom into one spike:

<img width="1440" alt="screen shot 2016-10-12 at 13 37 40" src="https://cloud.githubusercontent.com/assets/2151639/19308738/73db9490-9081-11e6-9214-92ac660eddca.png">

You can see the following pattern: MessageQueue.next(), a couple of intermediate methods that results into the actual View.draw call. I normal cases we shouldn't be able to see anything here, for example here is a trace at zoom 1 for 10 seconds:

<img width="1440" alt="screen shot 2016-10-12 at 13 44 21" src="https://cloud.githubusercontent.com/assets/2151639/19308859/0c49f384-9082-11e6-88bb-3a5824ede4b8.png">

**Comment by tobrun** [comment]
Been adding some log statements to the core code to see what is triggering the invalidations and noticed that the following in `Map::Impl::render` is called constantly:

```cpp
    } else if (painter->needsAnimation()) {
        updateFlags |= Update::Repaint;
        asyncUpdate.send();
    }
```

When starting a map not from zoom 0, this is only called once.

**Comment by tobrun** [comment]
Proposed solution in #6747

**Comment by kkaefer** [comment]
Fixed in #6979
